src/pages/docs/resources/typescript.md

```diff
@@ -0,0 +1,93 @@
+---
+title: TypeScript
+label: TypeScript
+layout: docs
+order: 4
+tocHeading: 2
+---
+
+# TypeScript
+
+Greenwood provide built-in support for TypeScript, either through type-stripping (default behavior) or with the ability to fallback to [using the TypeScript compiler](/docs/reference/configuration/#use-typescript-compiler) if you're leveraging certain transformation based TypeScript features (like [`enums` and `namespaces`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#the---erasablesyntaxonly-option)) or JavaScript syntax like Decorators. You can see an example repo [here](https://github.com/thescientist13/greenwood-native-typescript)
+
+> You can read [this guide](https://nodejs.org/en/learn/typescript/run-natively) to learn more about running TypeScript with NodeJS, including the [`--experimental-transform-types`](https://nodejs.org/docs/latest-v23.x/api/cli.html#--experimental-transform-types) flag.
+
+## Setup
+
+The below steps will help you get up and running with TypeScript in your Greenwood project. The general recommendation is to use type-stripping during development for faster live reload, and then run TypeScript during CI (e.g. GitHub Actions) to check and enforce all types, e.g. `tsc --project tsconfig.json`.
+
+1. You will need to use Node **>= 22.6.0** and set the `--experimental-strip-types` flag
+1. Install TypeScript into your project, e.g. `npm i typescript --save-dev`
+1. Create a _tsconfig.json_ file at the root of your project with these minimum configuration settings
```

Comment:
We should probably callout / link to the configuration docs for the `useTsc` fallback